src/components/CommentSummaryButton/index.tsx

```diff
@@ -0,0 +1,152 @@
+
+export type DefaultProps = {
+  createRef: () => React.RefObject<HTMLDivElement>;
+};
+
+type PropsFromState = {
+  comments: Comment[] | undefined;
+  showOverlay: boolean;
+};
+
+type Props = PublicProps & DefaultProps & PropsFromState & ConnectedReduxProps;
+
+export class CommentSummaryButtonBase extends React.Component<Props, State> {
+  private buttonRef = this.props.createRef();
+
+  static defaultProps = {
+    createRef: () => React.createRef<HTMLDivElement>(),
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.state = { canAccessRefs: false };
+  }
+
+  componentDidMount() {
+    this.setState({ canAccessRefs: true });
+  }
+
+  renderCommentSummary() {
+    const { comments, dispatch } = this.props;
+
+    return (
+      <>
+        <CommentSummary comments={comments || []} />
+        <div className={styles.summaryControls}>
+          <Button
+            onClick={() => {
+              dispatch(commentsActions.hideFinishReviewOverlay());
+            }}
+            size="sm"
+            variant="primary"
+          >
+            {gettext('Close')}
+          </Button>
+        </div>
+      </>
+    );
+  }
+
+  renderOverlay() {
+    const { dispatch, showOverlay } = this.props;
+    const { canAccessRefs } = this.state;
+
+    if (!canAccessRefs) {
+      return null;
+    }
+
+    if (!this.buttonRef.current) {
+      // This shouldn't be possible since canAccessRefs is set in
+      // componentDidMount() when refs are guaranteed to exist.
+      throw new Error('buttonRef.current was unexpectedly empty');
+    }
+
+    return (
+      <Overlay
+        onHide={() => {
+          dispatch(commentsActions.hideFinishReviewOverlay());
+        }}
+        placement="bottom"
+        rootClose
+        show={showOverlay}
+        target={this.buttonRef.current}
+      >
+        {({ arrowProps, placement, ref, style }) => {
+          return (
+            <Popover
+              arrowProps={arrowProps}
+              className={styles.popover}
+              id="finish-review-popover"
+              placement={placement}
+              ref={ref}
+              style={style}
+            >
+              <Popover.Title as="h3">{gettext('Comments')}</Popover.Title>
+              <Popover.Content>{this.renderCommentSummary()}</Popover.Content>
+            </Popover>
+          );
+        }}
+      </Overlay>
+    );
+  }
+
+  render() {
+    const { dispatch } = this.props;
+
+    return (
+      <>
+        <Button
+          onClick={() => {
+            // TODO: rename this to toggleSummaryOverlay()
+            // https://github.com/mozilla/addons-code-manager/issues/1166
+            dispatch(commentsActions.toggleFinishReviewOverlay());
+          }}
+          // This type has a conflicting definition. See:
+          // https://github.com/react-bootstrap/react-bootstrap/issues/4706
+          // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+          // @ts-ignore
```

Comment:
I thought the definition file was fixing this problem? 🤔

Reply:
This is a separate issue. I could add an overload to our local types to fix it but I'm actually not sure how to define it. It's tricky because it's a legacy ref and every time I try to add a type to satisfy the current def, I get a conflict with one of its union'd values. The erroris similar to one you'd use [discriminated unions](https://basarat.gitbooks.io/typescript/docs/types/discriminated-unions.html) to solve but I don't see how you could do that in this case. The ref is wrong at a deeper level within bootstrap and I don't think it's worth spending too much time on.

<details>
<summary>Example error</summary>

```

src/components/CommentSummaryButton/index.tsx:126:11 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: Readonly<ReplaceProps<"button", BsPrefixProps<"button"> & ButtonProps>>): Button<"button">', gave the following error.
    Type 'RefObject<HTMLDivElement>' is not assignable to type '(string & ((instance: HTMLButtonElement | null) => void)) | (string & RefObject<HTMLButtonElement>) | (((instance: Button<"button"> | null) => void) & ((instance: HTMLButtonElement | null) => void)) | ... 4 more ... | undefined'.
      Type 'RefObject<HTMLDivElement>' is not assignable to type 'RefObject<Button<"button">> & RefObject<HTMLButtonElement>'.
        Type 'RefObject<HTMLDivElement>' is not assignable to type 'RefObject<Button<"button">>'.
          Type 'HTMLDivElement' is missing the following properties from type 'Button<"button">': context, setState, forceUpdate, render, and 3 more.
  Overload 2 of 2, '(props: ReplaceProps<"button", BsPrefixProps<"button"> & ButtonProps>, context?: any): Button<"button">', gave the following error.
    Type 'RefObject<HTMLDivElement>' is not assignable to type '(string & ((instance: HTMLButtonElement | null) => void)) | (string & RefObject<HTMLButtonElement>) | (((instance: Button<"button"> | null) => void) & ((instance: HTMLButtonElement | null) => void)) | ... 4 more ... | undefined'.
      Type 'RefObject<HTMLDivElement>' is not assignable to type 'RefObject<Button<"button">> & RefObject<HTMLButtonElement>'.
        Type 'RefObject<HTMLDivElement>' is not assignable to type 'RefObject<Button<"button">>'.

126           ref={this.buttonRef}
              ~~~

  node_modules/@types/react/index.d.ts:104:9
    104         ref?: LegacyRef<T>;
                ~~~
    The expected type comes from property 'ref' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<Button<"button">> & Readonly<ReplaceProps<"button", BsPrefixProps<"button"> & ButtonProps>> & Readonly<{ children?: ReactNode; }>'
  node_modules/@types/react/index.d.ts:104:9
    104         ref?: LegacyRef<T>;
                ~~~
    The expected type comes from property 'ref' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<Button<"button">> & Readonly<ReplaceProps<"button", BsPrefixProps<"button"> & ButtonProps>> & Readonly<{ children?: ReactNode; }>'
```


</details>